### PR TITLE
[ttLib.removeOverlaps] Support CFF table

### PR DESCRIPTION
**Pull request by khaledhosny**
Part of #3512.

### CODE REVIEW

**Comment by behdad** [comment]
Thanks. We probably should add an option (default on?) to remove unused subroutines if anything was changed.

**Comment by behdad** [comment]
Please make the `remove_unused_subroutines` optional, since eg. `CFF2ToCFF` does it itself too...

**Comment by khaledhosny** [comment]
> Please make the `remove_unused_subroutines` optional, since eg. `CFF2ToCFF` does it itself too...

Done.

**Comment by behdad** [comment]
Please also add `*, ` before most of the options so they can only be called by name.

**Comment by khaledhosny** [comment]
> Please also add `*, ` before most of the options so they can only be called by name.

I can do that for the new functions, but wouldn’t doing so for the old ones potentially break user code?

**Comment by behdad** [comment]
> I can do that for the new functions,

Do it for new arguments.

**Comment by anthrotype** [comment]
could we add some tests in https://github.com/fonttools/fonttools/blob/main/Tests/ttLib/removeOverlaps_test.py ?

**Comment by khaledhosny** [comment]
> could we add some tests in https://github.com/fonttools/fonttools/blob/main/Tests/ttLib/removeOverlaps_test.py ?

I wondered about this as well, but the existing code does not have much tests either, so I thought it might be implicitly testes by the instancer, so this code would get some testing when it get integrated into the instancer as well.

**Comment by anthrotype** [comment]
oh you're right actually.

File: Lib/fontTools/ttLib/removeOverlaps.py [hunk 1]
```diff
@@ -225,27 +219,170 @@ def removeOverlaps(
...
+    log.debug("Removed overlaps for %s glyphs:\n%s", len(modified), " ".join(modified))
+
+
+def removeOverlaps(
+    font: ttFont.TTFont,
+    glyphNames: Optional[Iterable[str]] = None,
+    removeHinting: bool = True,
+    ignoreErrors: bool = False,
+    *,
+    removeUnusedSubroutines: bool = True,
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/removeOverlaps.py:285]: why would one wish to keep unused subroutines?

**Review by anthrotype — APPROVED** [review]

File: Lib/fontTools/ttLib/removeOverlaps.py (hunk 1 above)
Comment (khaledhosny) [reply Lib/fontTools/ttLib/removeOverlaps.py:285]: Behdad requested it, since code calling this might have removed unused subroutines already.

Comment (behdad) [reply Lib/fontTools/ttLib/removeOverlaps.py:285]: Yeah, in `CFF2ToCFF` we have done that already.

**Pull request merged**